core/src/main/java/com/yahoo/oak/EntrySet.java

```diff
@@ -887,9 +708,31 @@ boolean copyEntry(ValueBuffer tempValue, EntrySet<K, V> srcEntrySet, int srcEntr
                 entries,                        // this entries array
                 entryIdx2intIdx(destEntryIndex) + OFFSET.NEXT.value + 1, (FIELDS - 1));
 
+        assert valuesMemoryManager.isReferenceConsistent(getValueReference(destEntryIndex));
+
         // now it is the time to increase nextFreeIndex
         nextFreeIndex.getAndIncrement();
         numOfEntries.getAndIncrement();
         return true;
     }
+
+    boolean isEntrySetValidAfterRebalance() {
+        int currIndex = getHeadNextIndex();
+        int prevIndex = -1;
+        while (currIndex > getLastEntryIndex()) {
+            if (!isValueRefValidAndNotDeleted(currIndex)) {
+                return false;
+            }
+            if (!valuesMemoryManager.isReferenceConsistent(getValueReference(currIndex))) {
+                return false;
+            }
+            if ( prevIndex>0 && currIndex - prevIndex != FIELDS) {
```

Comment:
I think I need to revisit the styling rules validation tool because I've seen many deviations here from common styling that it didn't catch.
```suggestion
            if (prevIndex > 0 && currIndex - prevIndex != FIELDS) {
```

Reply:
This should be done in separate PR